core-modular/src/utilities/helpers/dirty-status-helpers.ts

```diff
@@ -0,0 +1,94 @@
+class DirtyStatusHelpersClass {
+  private storageKey = 'luigi.dirtyStore';
+  protected unsavedChanges!: Record<string, any>;
+
+  constructor() {
+    this.initChangesData();
+  }
+
+  initChangesData(): void {
+    const dirtyStore = this.parseDirtyStore();
+
+    if (dirtyStore) {
+      this.unsavedChanges = dirtyStore;
+      this.clearDirtyState();
+    } else {
+      this.unsavedChanges = {
+        isDirty: false,
+        persistUrl: null
+      };
+    }
+
+    this.storeDirtyStatus();
+  }
+
+  parseDirtyStore(): Record<string, any> | null {
```

Comment:
Storing the dirtyState in localstorage needs to be clarified with Philipp.
In core we hold the dirtyState only in memory, right? Or did I just miss that?
What happens if you have dirtyState in localstorage `true` and than replace the url with another microfrontend and press enter?